src/query/src/query_engine/state.rs

```diff
@@ -88,6 +89,8 @@ impl QueryEngineState {
         }
         analyzer.rules.insert(0, Arc::new(TypeConversionRule));
         analyzer.rules.insert(0, Arc::new(StringNormalizationRule));
+        Self::remove_analyzer_rule(&mut analyzer.rules, CountWildcardRule {}.name());
```

Comment:
I don't get the point here. Looks like it removed the `CountWildcardRule` and added it back again at the first position.  Why do that?

Reply:
Adjust the position. It can make sure this rule is always evaluated at first

Reply:
I remember there are other codes that did this adjustment before. Can we remove all the default rules, and organize them by ourselves? It would be more readable.

Reply:
Yes, just a few lines below. I'd prefer not to adjust the default ordering unless we have to. Rules always have implicit requirement/dependence on the ordering, and this part is not that carefully tested on either datafusion or ours. So follow the default ordering would be better